### PR TITLE
Add batch inference tutorial

### PR DESCRIPTION
**Pull request by orasimus**

### CODE REVIEW

File: source/tutorials/batch-inference/index.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+In this tutorial you will learn how to create and run a Batch Inference execution in Valohai. This execution will use TensorFlow 2.4.1 to run inference on ten unique images using a pre-trained MNIST model.
+
+.. admonition:: Prerequirements
+    :class: attention
+
+    For this tutorial you will need:
+
+    * Python 3
+
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:15]:
```suggestion
* ``pip install --upgrade valohai-cli``

```

File: source/tutorials/batch-inference/index.rst [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+To easily run our batch inference on Valohai, we will use it to run our code from the very beginning.
+
+If you don't already have a Valohai account, go to `<https://app.valohai.com/>`_ to create one for yourself.
+
+Similarly if you don't already have the Valohai command-line client, install it:
+
+.. code-block:: bash
+
+  pip install valohai-cli
+
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:40]: I'd remove this and just include it in the requirements, like in some of the other tutorials

```suggestion

```

File: source/tutorials/batch-inference/index.rst [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    images = [load_image(p) for p in image_paths]
+    results = [run_inference(i) for i in images]
+
+    with open(vh.outputs().path('results.json'), 'w') as f:
+        json.dump(results, f)
+
+If everything went according to plan, you can now preview the results in the Outputs-tab:
+
+.. image:: tensorflow-batch-inference-tutorial-2.png
+   :alt: Results of our batch inference execution
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:161]:
```suggestion
   :alt: Results of our batch inference execution
   
.. seealso::

    * `Valohai CLI </tutorials/valohai-cli/>`_
    * `Using Docker Images </topic-guides/docker-images/>>`_
    * `Valohai APIs </tutorials/apis/>`_
```

File: source/tutorials/batch-inference/index.rst [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    - step:
+        name: Batch Inference
+        image: tensorflow/tensorflow:2.4.1
+        command:
+        - pip install pillow valohai-utils
+        - python batch_inference.py
+        inputs:
+        - name: model
+            default: s3://valohai-public-files/tensorflow-batch-inference-tutorial/model.h5
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:71]:
```suggestion
          default: s3://valohai-public-files/tensorflow-batch-inference-tutorial/model.h5
```

File: source/tutorials/batch-inference/index.rst [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        - name: images
+            default: s3://valohai-public-files/tensorflow-batch-inference-tutorial/images.zip
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:73]:
```suggestion
          default: s3://valohai-public-files/tensorflow-batch-inference-tutorial/images.zip
```

File: source/tutorials/batch-inference/index.rst [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+.. code-block:: python
+
+    image_paths = vh.inputs('images').paths()
+
+    images = [load_image(p) for p in image_paths]
+    results = [run_inference(i) for i in images]
+
+    with open(vh.outputs().path('results.json'), 'w') as f:
+        json.dump(results, f)
+
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:157]:
```suggestion
Let’s run another execution on Valohai:

.. code-block:: bash
    vh exec run --adhoc "Batch Inference"
    
```

File: source/tutorials/batch-inference/index.rst [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    * a trained MNIST model
+    * some images to classify, packaged in a single zip file
+
+    To make it easy for you they are available here, no need to download them:
+
+    * Model: `<s3://valohai-public-files/tensorflow-batch-inference-tutorial/model.h5>`_
+    * Images: `<s3://valohai-public-files/tensorflow-batch-inference-tutorial/images.zip>`_
+
+    If you want to, you can train the required model by following the tutorial `here <https://www.tensorflow.org/tutorials/quickstart/beginner>`_.
```
Comment (DrazenDodik) [inline source/tutorials/batch-inference/index.rst:26]: If you want to, you can train the required model by following our `Valohai 101 tutorial </tutorials/quickstart/>`_.

**Review by DrazenDodik — CHANGES_REQUESTED** [review]
Looks good!  Added a couple of small comments.

I'd prefer if we tie this into the official Valohai quickstart, rather than the Tensorflow sample without any Valohai parts.

Did you try to predict with the model? What kind of results did you get? If the accuracy is as off as I'm seeing, then we should add a disclaimer that the model was trained with just a few epochs (like the TF sample goes through only 5 epochs)

**Comment by orasimus** [comment]
I just checked and it got 5/10 correct. One image is of an X, so any answer is wrong.

So I'd say the model is okay, the issue is more with my hand-drawn input images, which probably don't look much like the rest of the MNIST dataset.
